spec/mailers/tasks_mailer_spec.rb

```diff
@@ -0,0 +1,35 @@
+require "rails_helper"
+
+RSpec.describe TasksMailer, type: :mailer do
+  describe "task_completed" do
+    let!(:user) { create(:user) }
+    let!(:current_user) { user }
+    let!(:user1) { create(:user) }
+    let!(:workspace) { create(:workspace, user: user) }
+    let!(:project) { create(:project, workspace: workspace, users: [user]) }
+    let!(:task1) { create(:task, project: project) }
+    let!(:task) { create(:task, watchers: [user1]) }
+    let!(:mail) { TasksMailer.task_completed(@task, current_user) }
+
+    before do
+      sign_in current_user
```

Comment:
sign_in in mailer?

Reply:
mailer works independetly from controller

remove sign_in

you already passing user to mailer